exporter/kafkaexporter/factory.go

```diff
@@ -56,6 +56,24 @@ func WithTracesMarshalers(tracesMarshalers ...TracesMarshaler) FactoryOption {
 	}
 }
 
+// WithMetricsMarshalers adds metricsMarshalers.
+func WithMetricsMarshalers(metricsMarshalers ...MetricsMarshaler) FactoryOption {
```

Comment:
I don't see this being references anywhere within that would allow for these to ether be added by default or added via some configuration.

Can I ask how you'd want someone to use this in their exporter?

Reply:
@MovieStoreGuy I think the question for how to use these factory options predates this PR.  The WithTraceMarshalers() option existed before this PR.  I've attempted 2 methods to utilize the custom options:

1. pass in explicit options to NewFactory() in components.go.  However, this isn't supported with the collector-builder and I could no longer use it
2. create a custom exporter factory where the NewFactory() just delegates to kafkaexporter.NewFactory() with additional options

Taking a step back, are you aware of precedence for extending subcomponents of exporters/receivers?  At a high level, extensions for auth looked close, but when I looked deeper the requirements for high-level abstractions to exist in the collector didn't feel right either (e.g go.opentelemetry.io/collector/config/configauth).  I know you are aware #8272 which would have been less messy if custom marshalers could be more easily added.